test/tools/ossfuzz/protoToYul.cpp

```diff
@@ -178,14 +178,16 @@ void ProtoConverter::visit(BinaryOp const& _x)
 // New var numbering starts from x_10 until x_16
 void ProtoConverter::visit(VarDecl const& _x)
 {
-	m_output << "let x_" << ((_x.id() % 7) + 10) << " := ";
+	m_output << "let x_" << m_numLiveVars << " := ";
 	visit(_x.expr());
```

Comment:
What exactly happens in this visit? I'm just wondering if it's safe to visit it before the next 2 lines.

Reply:
The RHS of a variable declaration is visited. The next two lines ensure that the total number of new variable declarations in current scope and globally are incremented only after the RHS is visited.

Otherwise, the expression within RHS could reference a not-yet-declared variable like so

```
let x_11 := x_11 + 1
```

by incrementing it post visit, we ensure that only previously declared vars are referenced in the RHS.